cli/stage_executor.py

```diff
@@ -71,12 +73,30 @@ def run_stage(stage_name, hide_state_steps, hide_dynamic_steps, simple_output):
         hide_dynamic_steps (bool): don't show runtime generated steps
         simple_output (bool): use the minimal outputter
     """
+    if simple_output:
+        PP.NO_FORMAT = True
+
     mon = Monitor(not hide_state_steps, not hide_dynamic_steps)
     printer = SimplePrinter() if simple_output else StepListPrinter(False)
     mon.add_listener(printer)
-    mon.parse_stage(stage_name)
-    mon.start()
+    try:
+        mon.parse_stage(stage_name)
+    except RenderingException as ex:
+        # pylint: disable=E1101
+        if isinstance(ex, StateRenderingException):
+            PP.println(PP.bold("An error occurred while rendering one of the following states:"))
+            for state in ex.states:
+                PP.print(PP.cyan("    - {}".format(state)))
+                PP.println(" ({})".format("/srv/salt/{}".format(state.replace(".", "/"))))
+        else:
+            PP.println(PP.bold("An error occurred while rendering the stage file:"))
+            PP.println(PP.cyan("    {}".format(ex.stage_file)))
+        PP.println()
+        PP.println(PP.bold("Error description:"))
+        PP.println(PP.red(ex.pretty_error_desc_str()))
```

Comment:
The above block looks like it's the same as the block in start_stage() in monitor.py. Can these two be sanely put into a separate helper function?

Reply:
Yes, you're right. I'll refactor this to have a single location with the prints for this kind of exception.